src/main/java/Gertrude.java

```diff
@@ -0,0 +1,286 @@
+import java.util.Scanner;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class Gertrude {
+
+    public static final String HORIZONTAL_LINE = "____________________________________________________________";
+    public static final String[] KEYWORDS = {"bye", "list", "mark", "unmark", "todo", "deadline", "event"};
+    public static final String[] ONE_WORD_KEYWORDS = {"bye", "list"};
+    public static final String FILE_PATH = "data/gertrude.txt"; //"src/main/java/data/gertrude.txt";
+
+    public static void printHorizontalLine() {
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    public static void printIntroduction() {
+        System.out.println("Hello, I'm Gertrude.");
+        System.out.println("What can I do for you?");
+    }
+
+    public static void printGoodbyeMessage() {
+        System.out.println("Bye. Hope to see you again soon!");
+    }
+
+    public static void printList(ArrayList<Task> tasks) {
+        for (int i = 1; i <= tasks.size(); i++) {
+            System.out.print(i + ".");
+            tasks.get(i-1).printTask();
+        }
+        System.out.println("You have " + tasks.size() + " tasks in the list.");
+    }
+
+    public static void markTask(String[] lineInputArr, ArrayList<Task> tasks) {
+        int index = Integer.parseInt(lineInputArr[1]);
+        if (index < 1 || index > tasks.size()) {
+            System.out.println("That is not a valid index.");
+        } else if (lineInputArr[0].equals("mark")) {
+            tasks.get(index - 1).markDone();
+        } else {
+            tasks.get(index-1).markNotDone();
+        }
+    }
+
+    public static void addTask(Task task, String lineInput, ArrayList<Task> tasks) {
+        tasks.add(task);
+        System.out.println("added: " + lineInput);
+    }
+
+    public static void deleteTask(String[] lineInputArr, ArrayList<Task> tasks) {
+        try {
+            Task taskToRemove = tasks.get(Integer.parseInt(lineInputArr[1]) - 1);
+            System.out.println("Ok. I've removed this task: ");
+            taskToRemove.printTask();
+            tasks.remove(Integer.parseInt(lineInputArr[1])-1);
+            System.out.println("You now have " + tasks.size() + " tasks in the list.");
+        } catch (NumberFormatException e) {
+            System.out.println("That is not a valid number.");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("That is not a valid index.");
+        }
+    }
+
+    public static void addTodo(String[] lineInputArr, ArrayList<Task> tasks) {
+        String name = "";
+        for(int i = 1; i < lineInputArr.length; i++) {
+            name += lineInputArr[i] + " ";
+        }
+        Todo newTodo = new Todo(name);
+        addTask(newTodo, name, tasks);
+    }
+
+    public static void addDeadline(String[] lineInputArr, ArrayList<Task> tasks) throws GertrudeException, KeywordException {
```

Comment:
The methods addDeadline and addEvent share similar logic (such as iterating through the array and concatenating strings). This can be refactored to avoid duplication